src/parser/monk/brewmaster/normalizers/ExpelHarm.js

```diff
@@ -0,0 +1,43 @@
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+import SPELLS from 'common/SPELLS';
+import { GIFT_OF_THE_OX_SPELL_IDS } from '../constants';
+
+function isGotOxHeal(event) {
+  return event.type === 'heal' && GIFT_OF_THE_OX_SPELL_IDS.includes(event.ability.guid);
+}
+
+/**
+ * Expel Harm comes after all the orbs are consumed, and honestly its a
+ * huge pain in the ass. This normalizer puts it first instead.
+ */
+export default class ExpelHarm extends EventsNormalizer {
+  normalize(events) {
+    // forEach is supposed to be safe for mutation
+    //
+    // inb4 this introduces tons of bugs
+    events.forEach((event, idx) => {
+      if(event.type !== 'cast' || event.ability.guid !== SPELLS.EXPEL_HARM.id) {
+        return;
+      }
+
+      // quick check that EH is after ALL orbs
+      if(events[idx+1].timestamp === event.timestamp && isGotOxHeal(events[idx+1])) {
+        console.error('gotox orb after eh');
+      }
+      // find the start of run of orbs consumed
+      // safe --- game won't let you cast EH with 0 orbs
+      let i = idx;
+      while(events[i-1].timestamp === event.timestamp) {
```

Comment:
I think some kind of buffer would be helpful here. In this log for example https://wowanalyzer.com/report/t2ZK6pQvgLGqj7by/1-Mythic+MOTHER+-+Kill+(4:19)/14-Mantyke/events, I don't think the heal events from EH happen exactly at the same timestamp, e.g. here I think at least the last 2 of the 3 heals were triggered from EH, yet they're not on the same timestamp
![image](https://user-images.githubusercontent.com/5068584/50423325-24409c80-0854-11e9-8df0-3fda11babc26.png)

Reply:
good catch, i hadn't seen any so i didn't implement it but it is handled now :+1: